brud06-sp

brusdani opened the pull request:
Pilotní verze, základní struktura souborů a složek. Jako základ je využit domácí úkol z labu10 (registrační a přihlašovací formulář). Vytvořeny základní třídy.

brusdani committed:
brud06-sp

Pilotní verze, základní struktura souborů a složek.
Jako základ je využit domácí úkol z labu10 (registrační a přihlašovací formulář). Vytvořeny základní třídy.

brusdani committed:
sp-update

Users are now able to register and to login.
Character selection and character creation is now working as well.

brusdani committed:
Quests and demo encounter added

Quests have been added. Basic encounter with monster at the end of the quest added.

brusdani committed:
Back at it

Character is now updated after each completed quest

brusdani committed:
Blacksmith feature and buying items

Added blacksmith store and option to buy items for gold. Character's inventory is now also being displayed.
Note: Selling items yet to be done

brusdani committed:
Equipment update

It is now possible to equip purchased equipment on your hero.
Also added possibility to delete own character.

brusdani committed:
Basic admin interface added

Added basic admin interface, ability for admin to create ingame items. Also different flow paths for admin and a player. Additionally selling items is now possible.

brusdani committed:
Visual update

Removed dexterity attribute because it had no use. Added timestamp which I plan to use for stamina recovery.
Additionally, default images added for character creation and monsters. EncounterDisplay now looks prettier.

brusdani committed:
Stamina feature added

Added depletion of stamina and automatic recovery. Furthermore stats from equipment are now added to characters' stats

brusdani committed:
Dungeon feature added

It is now possible to fight monsters in dungeon, still need to add progress tracker, but now it's time for bedge.

brusdani committed:
Admin update

Admin functions improved and added. Added ability to ban users. Added pages to edit ingame items.
This also means that core functions are completed and only polishing is required.

brusdani commented:
Aplikace ve verzi pro feedback před obhajobou je dostupná na: https://eso.vse.cz/~brud06/sp/

**Testovací Účty**

Admin účet:
- Email: `admin@admin.cz`
- Password: `admin123`

Postava připravená s goldy pro vyzkoušení funkce dungeonů a blacksmitha:
- Email: `pikachu@pokemons.czk`
- Password: `12345678`

brusdani committed:
Inventory update and code polishing

Added restrictions to player's inventory(handle cases with full inventory capacity and trying to buy item, that's already owned) and made slight changes in code structure.

nvbach91 commented:
- stranka character creation by mel byt disabled pro uzivatele ktery uz ma vytvorenou postavu
- pokud hrac vstoupil do dungeon nesmi ho opustit dokud ho nedokonci nebo zemre
- tu interakci s questy a prisery by se melo trochu obohatit napr. na bazi tahu hrac utoci, hrac brani, prisera utoci, prisera brani atd. jinak nevidim smysl kupovat itemy

nvbach91 commented on www/brud06/sp/classes/Monster.php:
```diff
@@ -0,0 +1,102 @@
+            list($this->id, $this->name, $this->level, $this->strength, $this->hitpoints, $this->luck, $this->isDungeonMonster) = $args;
+        }
+    }
+    public function getId()
```
tady si opakujete gettery a settery, co treba myslet na magic methods `__get` a `__set` pro jednoduche settery a gettery?

nvbach91 commented on www/brud06/sp/db/CharactersDB.php:
```diff
@@ -0,0 +1,93 @@
+class CharactersDB extends Database
+{
+
+    function createCharacter($character)
```
v nazvech techle metod by nemelo zaznit slovo `character` protoze to uz je z kontextu jasne

nvbach91 commented on www/brud06/sp/db/QuestsDB.php:
```diff
@@ -0,0 +1,34 @@
+
+    function getRandomQuests($limit)
+    {
+        $results = $this->runQuery("SELECT * FROM sp_quests ORDER BY RAND() LIMIT :limit", ['limit' => $limit]);
```
udelane questy by se nemely znovu ukazovat?

brusdani replied:
Tohle je zrovna úmyslné. Kdyby došly questy z databáze, hráč by neměl jak získávat xp a goldy.

nvbach91 commented on www/brud06/sp/db/database.php:
```diff
@@ -0,0 +1,68 @@
+<?php
+
+const DB_HOSTNAME = 'localhost';
```
tohle dejte do samostatneho souboru `config.php` nebo `.env` a chrante ho pomocu .htaccess

brusdani commented:
> * stranka character creation by mel byt disabled pro uzivatele ktery uz ma vytvorenou postavu
> * pokud hrac vstoupil do dungeon nesmi ho opustit dokud ho nedokonci nebo zemre
> * tu interakci s questy a prisery by se melo trochu obohatit napr. na bazi tahu hrac utoci, hrac brani, prisera utoci, prisera brani atd. jinak nevidim smysl kupovat itemy

Dobrý den,
můžete mi prosím více rozepsat bod 3 s tou interakcí? Úplně nerozumím, co je ode mě očekáváno.
Myslíte tím něco na způsob animací, kde postupně hráči i příšeře mizí životy?

nvbach91 commented:
ne, zadna animace, myslim tim pro uzivatele treba tlacitko utocit, tlacitko branit, tlacitko pass. tyhle akce by takt delaly prisery nejak nahodne podle stavu HP

brusdani commented:
> ne, zadna animace, myslim tim pro uzivatele treba tlacitko utocit, tlacitko branit, tlacitko pass. tyhle akce by takt delaly prisery nejak nahodne podle stavu HP

Bohužel asi nebudu schopen překopat celý soubojový systém za jeden či dva dny. Naprogramovat správně tahovku není záležitost na pár hodin a nemám na to ani vymyšlený plán.
Hra momentálně dělá to, co jsem definoval při návrhu projektu v issue, tudíž, že souboje simuluje a hráč nemůže jejich průběh nijak ovlivnit.
Pointa kupování itemů je jednoduchá - na poražení silnějších monster (silnější = více statů), je potřeba mít více statů.
Chápu, že tento styl gameplaye není nijak extra zajímavý a není pro každého. Ale i tak mají hry tohoto typu své publikum.
Můj projekt byl velmi inspirován hrou Shakes and Fidget.
Zde odkaz:
https://www.youtube.com/watch?v=C5dMrb9Pdu0&ab_channel=Strambic94
Šlo by vymyslet na úpravu něco jiného?

edit: Ostatní připomínky budou opraveny

nvbach91 commented:
nemusi to byt tahovka, to bylo jen doporuceni, muzete jeste napriklad rozepsat detailne jak probihal souboj, davat hracum treba nejaky report souboje

brusdani committed:
Battle log update

Battle log feature added for a player.
Character creation bug fix - it is no longer available when character's already been created.
Config.php file and .htaccess added

brusdani committed:
Added magic methods

__set and __get methods added for all classes to make getters and setter easier in the future
old getters still kept in place for now in order to avoid complete rewrite.
Additionally, after completing dungeon floor player stays within the dungeon.

nvbach91 commented:
- upozorneni na zvyseni levelu
- admin nema se podle ceho rozhodnout jeslti zabanovat hrace?
- nahravani obrazku na server a ulozit ho
- oauth?

brusdani committed:
Obhajoba

Update z obhajoby:
Added GitHub oauth, level up alert, admin now sees the users' characters.
Admin can now also upload images into server.
